types/react-native/index.d.ts

```diff
@@ -7480,6 +7595,23 @@ export interface LinkingStatic extends NativeEventEmitter {
     sendIntent(action: string, extras?: Array<{ key: string; value: string | number | boolean }>): Promise<void>;
 }
 
+export interface LogBoxStatic {
+    /**
+     * Silence any logs that match the given strings or regexes.
+     */
+    ignoreLogs(patterns: (string | RegExp)[]): void;
+
+    /**
+     * Toggle error and warning notifications
+     * Note: this only disables notifications, uncaught errors will still open a full screen LogBox.
+     * @param ignore whether to ignore logs or not
+     */
+    ignoreAllLogs(ignore: boolean): void;
```

Comment:
For this function technically the original flow types made `ignore` optional and the code does `!!ignore` to make sure it's a bool. I decided to make it required because if you used `LogBox.ignoreAllLogs()` without a param it would default to `false` and enable logs, which is the opposite to what you would expect reading the code.

Reply:
Nice catch, fixed with https://github.com/facebook/react-native/pull/29310, we can update the type after that lands or we can land the types assuming it's fixed

Reply:
@rickhanlonii Happy to do either, I guess it depends on when the release will be patched

Reply:
I agree with @rickhanlonii, regardless of when it’s released upstream.

Reply:
The change landed, not sure when the next pick and release to 0.63 will be

Reply:
@alloy which of @rickhanlonii's options do you agree with? Changing to this now, or later?
```suggestion
    ignoreAllLogs(ignore?: boolean): void;
```

Reply:
Apologies, should’ve been more clear. I agree with **not** making it optional in the TS typings.